aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -988,7 +989,7 @@ Tensor embedding_dense_backward_mps(const Tensor& grad_,
 
   auto inputType = getMPSDataType(self);
   auto sourceType = getMPSDataType(source);
-  if (inputType == MPSDataTypeUInt8 || (!is_macos_13_or_newer() && inputType == MPSDataTypeBool)) {
+  if (inputType == MPSDataTypeUInt8 || inputType == MPSDataTypeBool) {
```

Comment:
If we are removing these conditions, we should raise a warning saying that Boolean won't work correctly with MacOS12 and should ask them to upgrade to MacOS13 ?

Reply:
No, it will work correctly everywhere (though alas I have no way of testing it right now), this change is needed to hide regression in MaOS 14, where it started to exhibit the same failures as MacOS 12.